src/modules/preferences.js

```diff
@@ -3,242 +3,262 @@
-        }
-      });
+      let promise = OS.File.read(filename); // Read the complete file as an array
+      let decoder = new TextDecoder();
 
-      let keyringControl = this._environment.get('GNOME_KEYRING_CONTROL');
-      PassFF.Preferences._gpgAgentEnv.push('GNOME_KEYRING_CONTROL=' + keyringControl);
+      let success = function(array) {
+        let re = /^([^=]+=[^;]+)/;
+        log.info('Retrieve Gpg agent variable from file');
 
-      log.debug('Set Gpg agent variable:', PassFF.Preferences._gpgAgentEnv);
+        PassFF.Preferences._gpgAgentEnv = [];
+        decoder.decode(array).split('\n').forEach(function(line) {
+          if(re.test(line)) {
+            PassFF.Preferences._gpgAgentEnv.push(re.exec(line)[0]);
+          }
+        });
 
-      return Promise.resolve('OK');
-    }.bind(this);
+        let keyringControl = this._environment.get('GNOME_KEYRING_CONTROL');
+        PassFF.Preferences._gpgAgentEnv.push('GNOME_KEYRING_CONTROL=' + keyringControl);
 
-    let error = function(reason) {
-      log.info('Can\'t read file. Getting gpg-agent variable from environment');
-      PassFF.Preferences._gpgAgentEnv = [
-        'GPG_AGENT_INFO=' + this._environment.get('GPG_AGENT_INFO'),
-        'GNOME_KEYRING_CONTROL=' + this._environment.get('GNOME_KEYRING_CONTROL')
-      ];
-      log.debug('Set gpg-agent variable :', PassFF.Preferences._gpgAgentEnv);
-      return Promise.resolve('OK');
-    }.bind(this);
+        log.debug('Set Gpg agent variable:', PassFF.Preferences._gpgAgentEnv);
 
-    promise = promise.then(success, error);
+        return Promise.resolve('OK');
+      }.bind(this);
 
-    promise.catch(function onError(reason) {
-      log.error('Failed to set gpg-agent variable', reason);
-    });
-  },
+      let error = function(reason) {
+        log.info('Can\'t read file. Getting gpg-agent variable from environment');
+        PassFF.Preferences._gpgAgentEnv = [
+          'GPG_AGENT_INFO=' + this._environment.get('GPG_AGENT_INFO'),
+          'GNOME_KEYRING_CONTROL=' + this._environment.get('GNOME_KEYRING_CONTROL')
+        ];
+        log.debug('Set gpg-agent variable :', PassFF.Preferences._gpgAgentEnv);
+        return Promise.resolve('OK');
+      }.bind(this);
 
-  get passwordInputNames() {
-    log.error("aaaaa ")
-    return this._params.passwordInputNames.split(',');
-  },
+      promise = promise.then(success, error);
 
-  get loginInputNames() {
-    return this._params.loginInputNames.split(',');
-  },
+      promise.catch(function onError(reason) {
+        log.error('Failed to set gpg-agent variable', reason);
+      });
+    },
 
-  get loginFieldNames() {
-    return this._params.loginFieldNames.split(',');
-  },
+    get passwordInputNames() {
+      log.error("aaaaa ")
+      return this._params.passwordInputNames.split(',');
+    },
 
-  get passwordFieldNames() {
-    return this._params.passwordFieldNames.split(',');
-  },
+    get loginInputNames() {
```

Comment:
Personally I think that those files are scoped like modules, but I'm not really sure about that, as I have just been writing smaller patches myself. 
Also, I don't think that `let` should ever leak into the global scope - Mozilla states "At the top level of programs and functions, let, unlike var, does not create a property on the global object."
@jvenant, do you have more input on that?